draco/analysis/delay.py

```diff
@@ -887,40 +982,44 @@ def delay_spectrum_gibbs(
 
     spec = []
 
-    total_freq = N // 2 + 1
+    total_freq = N if complex_input else N // 2 + 1
 
     if fsel is None:
         fsel = np.arange(total_freq)
 
     # Construct the Fourier matrix
-    F = fourier_matrix_r2c(N, fsel)
+    F = fourier_matrix_c2c(N, fsel) if complex_input else fourier_matrix_r2c(N, fsel)
 
     # Construct a view of the data with alternating real and imaginary parts
     data = data.astype(np.complex128, order="C").view(np.float64).T.copy()
 
     # Window the frequency data
-    if window:
+    if window is not None:
 
         # Construct the window function
         x = fsel * 1.0 / total_freq
-        w = window_generalised(x, window="nuttall")
+        w = window_generalised(x, window=window)
         w = np.repeat(w, 2)
 
         # Apply to the projection matrix and the data
         F *= w[:, np.newaxis]
         data *= w[:, np.newaxis]
 
-    is_real_freq = (fsel == 0) | (fsel == N // 2)
+    if complex_input:
+        is_real_freq = np.zeros_like(fsel).astype(bool)
+    else:
+        is_real_freq = (fsel == 0) | (fsel == N // 2)
 
     # Construct the Noise inverse array for the real and imaginary parts (taking
-    # into account that the zero and Nyquist frequencies are strictly real)
+    # into account that the zero and Nyquist frequencies are strictly real if the delay
+    # spectrum is assumed to be real)
     Ni_r = np.zeros(2 * Ni.shape[0])
     Ni_r[0::2] = np.where(is_real_freq, Ni, Ni / 2**0.5)
     Ni_r[1::2] = np.where(is_real_freq, 0.0, Ni / 2**0.5)
 
     # Create the Hermitian conjugate weighted by the noise (this is used multiple times)
-    FTNih = F.T * Ni_r[np.newaxis, :] ** 0.5
-    FTNiF = np.dot(FTNih, FTNih.T)
+    FTNih = F.T.conj() * Ni_r[np.newaxis, :] ** 0.5
+    FTNiF = np.dot(FTNih, FTNih.T.conj())
```

Comment:
Given the comments above, I'm not 100% sure that this is actually what we want to be doing in here.